libs/agent-sm/bootstrap/src/main/java/org/opensearch/javaagent/bootstrap/AccessController.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.javaagent.bootstrap;
+
+/**
+ * Utility class to run code in a privileged block.
+ */
+public final class AccessController {
+    /**
+     * Don't allow instantiation an {@code AccessController}
+     */
+    private AccessController() {}
+
+    /**
+     * Performs the specified action in a privileged block.
+     *
+     * <p> If the action's {@code run} method throws an (unchecked)
+     * exception, it will propagate through this method.
+     *
+     * @param <T> the type of the value returned by the PrivilegedAction's
+     *                  {@code run} method
+     *
+     * @param action the action to be performed
+     *
+     * @return the value returned by the action's {@code run} method
+     */
+    public static <T> T doPrivileged(PrivilegedAction<T> action) {
+        T result = action.run();
+        return result;
+    }
+
+    private static PrivilegedActionException wrapException(Exception e) {
+        return new PrivilegedActionException(e);
+    }
+
+    /**
+     * Performs the specified action.
+     *
+     * <p> If the action's {@code run} method throws an <i>unchecked</i>
+     * exception, it will propagate through this method.
+     *
+     * @param <T> the type of the value returned by the
+     *                  PrivilegedExceptionAction's {@code run} method
+     *
+     * @param action the action to be performed
+     *
+     * @return the value returned by the action's {@code run} method
+     *
+     * @throws    PrivilegedActionException if the specified action's
+     *         {@code run} method threw a <i>checked</i> exception
+     */
+    public static <T> T doPrivileged(PrivilegedExceptionAction<T> action) throws PrivilegedActionException {
+        try {
+            T result = action.run();
+            return result;
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            throw wrapException(e);
```

Comment:
Why do we need to wrap `Exception`? We don't wrap `RuntimeException`.

Reply:
Removed `wrapException` and created a new `PrivilegedActionException` inline which is what `wrapException was for and this was its single usage.

Reply:
But why wrap it at all? If my code throws an IOException, I'll instead get a PrivilegedActionException with an IOException as the cause. Why not just throw the IOException?

Reply:
Because it fits the method signature. Callers of this method expect it to throw a PrivilegedActionException

Reply:
Without it, then we need to change the method signature to throw a generic exception. Unless you know what specific exception could be thrown within the runnable block, then anyone using this existing signature needs to update their `catch` blocks to catch a generic Exception.

Reply:
Oh yes, of course. However, this reminds me of my main complaint about this interface. Let me start a new comment thread though.